src/main/java/org/springframework/data/redis/connection/RedisConfiguration.java

```diff
@@ -135,6 +144,32 @@ static RedisPassword getPasswordOrElse(@Nullable RedisConfiguration configuratio
 		return isPasswordAware(configuration) ? ((WithPassword) configuration).getPassword() : other.get();
 	}
 
+	/**
+	 * @param configuration can be {@literal null}.
+	 * @param other a {@code Supplier} whose result is returned if given {@link RedisConfiguration} is not
+	 *          {@link #isHostAndPortAware(RedisConfiguration) port aware}.
+	 * @return never {@literal null}.
+	 * @throws IllegalArgumentException if {@code other} is {@literal null}.
+	 */
+	static Integer getPortOrElse(@Nullable RedisConfiguration configuration, Supplier<Integer> other) {
```

Comment:
This could be `int` and we could change `Supplier<Integer>` to `IntSupplier`.

Reply:
I used Integer just for consistency with getDatabaseOrElse which also return an integer and use a Supplier<Integer>. Do you want me to change both, just this new method or keep as it is?

Reply:
Using primitives removes the possibility of running into `null`. Let's change this one and we will have another look during the merge.

Reply:
Done, let me know if you think I need to change something else, thanks